cc/defs.bzl

```diff
@@ -34,6 +34,14 @@ def _common_c_opts(nocopts, pedantic = False):
 def _construct_local_includes(local_includes):
     return [construct_local_include(path) for path in local_includes]
 
+def _default_features():
+    return select({
+        # treat_warnings_as_errors passes the option -fatal-warnings
+        # to the linker which ld on mac does not understand.
+        "@platforms//os:macos": [],
```

Comment:
I discovered this issue when trying to build my branch on mac.

Seems like it's something that will need to get patched upstream.

We're going to be using our own implementation shortly though so we will also be able to fix it ourselves to as to not have to wait on it to land.